projects/e2fsprogs/fuzz/ext2fs_image_read_write_fuzzer.cc

```diff
@@ -0,0 +1,86 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <stddef.h>
+#include <stdint.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <fuzzer/FuzzedDataProvider.h>
+
+#include "ext2fs/ext2fs.h"
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+
+  enum Fuzzer {
+    ext2fsImageBitmapRead,
+    ext2fsImageInodeRead,
+    ext2fsImageSuperRead,
+    ext2fsImageBitmapWrite,
+    ext2fsImageInodeWrite,
+    ext2fsImageSuperWrite,
+    kMaxValue = ext2fsImageSuperWrite
+  };
+
+  FuzzedDataProvider stream(data, size);
+  const Fuzzer f = stream.ConsumeEnum<Fuzzer>();
+  static const char* fname = "/dev/shm/ext2_test_file";
+
+  // Write our data to a temp file.
+  int fd = open(fname, O_RDWR|O_CREAT|O_TRUNC);
+  std::vector<char> buffer = stream.ConsumeRemainingBytes<char>();
+  write(fd, buffer.data(), buffer.size());
+  close(fd);
+
+  ext2_filsys fs;
+  errcode_t retval = ext2fs_open(
+      fname,
+      0, 0, 0,
+      unix_io_manager,
+      &fs);
+
+  if (!retval) {
+    switch (f) {
+      case ext2fsImageBitmapRead: {
+        ext2fs_image_bitmap_read(fs, fd, 0);
+        break;
+      }
+      case ext2fsImageInodeRead: {
+        ext2fs_image_inode_read(fs, fd, 0);
+        break;
+      }
+      case ext2fsImageSuperRead: {
+        ext2fs_image_super_read(fs, fd, 0);
+        break;
+      }
+      case ext2fsImageBitmapWrite: {
+        ext2fs_image_bitmap_write(fs, fd, 0);
+        break;
+      }
+      case ext2fsImageInodeWrite: {
+        ext2fs_image_inode_write(fs, fd, 0);
+        break;
+      }
+      case ext2fsImageSuperWrite: {
+        ext2fs_image_super_write(fs, fd, 0);
+        break;
+      }
+      default: {
+        return 0;
```

Comment:
can you put an assert or abort here please? so that if we ever hit this branch we fail loudly and fix the fuzz target